tests/aws/services/events/test_events_cross_account_region.py

```diff
@@ -0,0 +1,268 @@
+                    "Action": "sqs:SendMessage",
+                    "Resource": queue_arn_secondary,
+                }
+            ],
+        }
+        aws_client_secondary.sqs.set_queue_attributes(
+            QueueUrl=queue_url_secondary,
+            Attributes={"Policy": json.dumps(policy_events_sqs_secondary)},
+        )
+
+        # Create rule in primary region
+        rule_name = f"test-rule-primary-sqs-{short_uid()}"
+        aws_client_primary.events.put_rule(
+            Name=rule_name,
+            EventPattern=json.dumps({"source": [SOURCE_PRIMARY]}),
+            # EventPattern=json.dumps({"source": [SOURCE_PRIMARY], **TEST_EVENT_PATTERN_NO_SOURCE}),
+            EventBusName=event_bus_name_primary,
+        )
+
+        # Create target in primary region sqs
+        target_id_sqs_primary = f"test-target-primary-sqs-{short_uid()}"
+        aws_client_primary.events.put_targets(
+            Rule=rule_name,
+            EventBusName=event_bus_name_primary,
+            Targets=[
+                {
+                    "Id": target_id_sqs_primary,
+                    "Arn": queue_arn_primary,
+                }
+            ],
+        )
+
+        # Create permission for event bus in primary region to send events to event bus in secondary region
+        role_name_bus_primary_to_bus_secondary = (
+            f"event-bus-primary-to-secondary-role-{short_uid()}"
+        )
+        assume_role_policy_document_bus_primary_to_bus_secondary = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {"Service": "events.amazonaws.com"},
+                    "Action": "sts:AssumeRole",
+                }
+            ],
+        }
+
+        role_arn_bus_primary_to_bus_secondary = aws_client_primary.iam.create_role(
+            RoleName=role_name_bus_primary_to_bus_secondary,
+            AssumeRolePolicyDocument=json.dumps(
+                assume_role_policy_document_bus_primary_to_bus_secondary
+            ),
+        )["Role"]["Arn"]
+
+        policy_name_bus_primary_to_bus_secondary = (
+            f"event-bus-primary-to-secondary-policy-{short_uid()}"
+        )
+        policy_document_bus_primary_to_bus_secondary = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": "",
+                    "Effect": "Allow",
+                    "Action": "events:PutEvents",
+                    "Resource": "arn:aws:events:*:*:event-bus/*",
+                }
+            ],
+        }
+
+        aws_client_primary.iam.put_role_policy(
+            RoleName=role_name_bus_primary_to_bus_secondary,
+            PolicyName=policy_name_bus_primary_to_bus_secondary,
+            PolicyDocument=json.dumps(policy_document_bus_primary_to_bus_secondary),
+        )
+
+        if is_aws_cloud():
+            time.sleep(10)
```

Comment:
I'm guilty of using this as well, but I think current best practice is to use retry

Reply:
This wait time is particularly for waiting for IAM propagation, I am not aware of a retry method that could handle this.